modules/core/keeper/msg_server_test.go

```diff
@@ -924,3 +924,140 @@ func (suite *KeeperTestSuite) TestChannelUpgradeTry() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestChannelUpgradeTimeout() {
+	var (
+		path *ibctesting.Path
+		msg  *channeltypes.MsgChannelUpgradeTimeout
+	)
+
+	cases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		{
+			name:     "success",
+			malleate: func() {},
+			expErr:   nil,
+		},
+		{
+			name: "success with error receipt",
+			malleate: func() {
+				errReceipt := channeltypes.ErrorReceipt{
+					Sequence: 0,
+					Message:  "error",
+				}
+				path.EndpointB.Chain.App.GetIBCKeeper().ChannelKeeper.SetUpgradeErrorReceipt(path.EndpointB.Chain.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, errReceipt)
+
+				suite.Require().NoError(path.EndpointB.UpdateClient())
+				suite.Require().NoError(path.EndpointA.UpdateClient())
+
+				upgradeErrorReceiptKey := host.ChannelUpgradeErrorKey(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+				errorReceiptProof, proofHeight := path.EndpointB.QueryProof(upgradeErrorReceiptKey)
+				counterpartyChannel := path.EndpointB.GetChannel()
+				channelProof, _, _ := path.EndpointA.QueryChannelUpgradeProof()
+
+				msg.PreviousErrorReceipt = &errReceipt
+				msg.ProofErrorReceipt = errorReceiptProof
+				msg.CounterpartyChannel = counterpartyChannel
+				msg.ProofChannel = channelProof
+				msg.ProofHeight = proofHeight
+			},
+			expErr: nil,
+		},
+		{
+			name: "invalid proof",
+			malleate: func() {
+				msg.ProofErrorReceipt = []byte("invalid proof")
+			},
+			expErr: commitmenttypes.ErrInvalidProof,
+		},
+		{
+			name: "invalid error receipt sequence",
```

Comment:
why is this invalid?

Reply:
It might be good to update the successful error receipt test case to indicate the error receipt is for a past upgrade and thus is stale

Reply:
yes, exactly, this is the usecase where we should actually be using `UpgradeCancel`